File: app/src/main/java/com/nedaluof/qurany/ui/myreciters/MyRecitersAdapter.java
package com.nedaluof.qurany.ui.myreciters;

import android.content.Context;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.nedaluof.qurany.R;
import com.nedaluof.qurany.data.model.Reciter;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

import javax.inject.Inject;

/**
 * Created by nedaluof on 7/7/2020.
 */
public class MyRecitersAdapter extends BaseQuickAdapter<Reciter, BaseViewHolder> {
    private MyRecitersView handleClick;
    @Inject
    Context context;

    public MyRecitersAdapter(int layoutResId, @Nullable List<Reciter> data, MyRecitersView handleClick) {
        super(layoutResId, data);
        this.handleClick = handleClick;
    }

    @Override
    protected void convert(@NotNull BaseViewHolder holder, Reciter reciter) {
        holder.setText(R.id.tv_reciter_name, reciter.getName().toUpperCase())
                .setText(R.id.tv_reciter_rewaya, reciter.getRewaya().toUpperCase())
                .setText(R.id.tv_reciter_sura_count, "There " + reciter.getCount() + " Sura")
                .getView(R.id.reciter_data_layout).setOnClickListener(v -> handleClick.onClickGetReciterData(reciter));

        holder.setImageResource(R.id.img_add_favorite, R.drawable.ic_favorite_selected);
        holder.getView(R.id.img_add_favorite)
                .setOnClickListener(v -> {
                    holder.setImageResource(R.id.img_add_favorite, R.drawable.ic_favorite_navigation);
                    handleClick.onClickDeleteFromMyReciters(reciter);
                });

    }

}

File: app/src/main/java/com/nedaluof/qurany/ui/myreciters/MyRecitersAdapter.kt
package com.nedaluof.qurany.ui.myreciters

import android.view.LayoutInflater
import android.view.ViewGroup
import com.nedaluof.qurany.data.model.Reciter
import com.nedaluof.qurany.databinding.ItemMyReciterBinding
import com.nedaluof.qurany.ui.base.BaseRecyclerAdapter

/**
 * Created by nedaluof on 12/12/2020.
 */
class MyRecitersAdapter(
    val onReciterClicked: (Reciter) -> Unit,
    val onDeleteFromMyRecitersClicked: (Reciter) -> Unit,
) : BaseRecyclerAdapter<Reciter>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
        MyRecitersVH(
            ItemMyReciterBinding.inflate(
                LayoutInflater.from(parent.context), parent,
                false
            )
        )

    inner class MyRecitersVH(
        val binding: ItemMyReciterBinding,
    ) : BaseRecyclerAdapter.BaseViewHolder(binding) {
        override fun onBind(position: Int) {
            val reciterData = items[position]
            with(binding) {
                myRecitersItemLyt.setOnClickListener { onReciterClicked(reciterData) }
                deleteReciterFromFavorite.setOnClickListener {
                    onDeleteFromMyRecitersClicked(
                        reciterData
                    )
                }
                reciter = reciterData
                executePendingBindings()
            }
        }
    }
}
